de.fhdo.lemma.servicedsl.extractor/src/de/fhdo/lemma/servicedsl/extractor/ServiceDslExtractor.xtend

```diff
@@ -0,0 +1,202 @@
+package de.fhdo.lemma.servicedsl.extractor
+
+import de.fhdo.lemma.data.ComplexType
+import de.fhdo.lemma.data.PrimitiveType
+import de.fhdo.lemma.service.Endpoint
+import de.fhdo.lemma.service.Import
+import de.fhdo.lemma.service.ImportType
+import de.fhdo.lemma.service.ImportedType
+import de.fhdo.lemma.service.Interface
+import de.fhdo.lemma.service.Microservice
+import de.fhdo.lemma.service.MicroserviceType
+import de.fhdo.lemma.service.Operation
+import de.fhdo.lemma.service.Parameter
+import de.fhdo.lemma.service.ServiceModel
+import de.fhdo.lemma.service.Visibility
+import de.fhdo.lemma.technology.CommunicationType
+import de.fhdo.lemma.technology.ExchangePattern
+import de.fhdo.lemma.technology.Protocol
+import de.fhdo.lemma.service.ImportedServiceAspect
+
+class ServiceDslExtractor {
+    static val IMPORTED_TECH_ALIASES = <String>newArrayList()
+    static val ID_PATTERN = "(\\^?)([a-zA-Z_])\\w*"
+    static val QUALIFIED_NAME_PATTERN = '''«ID_PATTERN»(\.«ID_PATTERN»)*'''
+    static val QUALIFIED_NAME_WITH_AT_LEAST_ONE_LEVEL_PATTERN = '''«ID_PATTERN»\.''' +
+        QUALIFIED_NAME_PATTERN
+
+    /**
+     * Extract ServiceModel
+     */
+    def String extractToString(ServiceModel serviceModel) {
+        val imports = <String>newArrayList
+        serviceModel.imports.forEach[imports.add(it.generate)]
+        val importStatements = if (!imports.empty)
+                String.join("\n", imports) + "\n\n"
+            else
+                ""
+
+        val microservices = String.join("\n\n", serviceModel.microservices.map[generate])
+
+        return '''«importStatements»«microservices»'''
+    }
+
+    private def generate(Import ^import) {
+        val importTypeKeyword = switch(^import.importType) {
+            case DATATYPES: "datatypes"
+            case MICROSERVICES: "microservices"
+            case TECHNOLOGY: "technology"
+            default: null
+        }
+        //if it is a technology add it to the names list for later use in the generation
+        if(importTypeKeyword === "technology") IMPORTED_TECH_ALIASES.add(^import.name)
+
+        return '''import «importTypeKeyword» from "«^import.importURI»" as «^import.name»'''
+    }
+
+    private def generate(Microservice service) {
+        val techImports = generateTechAnnotation(service)
+        val preamble = '''«service.visibility.generate» «service.type.generate»'''
+        val hasInterfaceOperations = service.interfaces.exists[!operations.empty]
+        return '''
+        «techImports»
```

Comment:
Why not invoking the method for generating the imports in the guillemets here, thereby rendering the variable redundant?